### PR TITLE
upgrade to ruby 3.1.4

### PR DESCRIPTION
**Pull request by wendelfabianchinsamy**
## Purpose

closes: https://github.com/datacite/sashimi/issues/175

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: app/models/heartbeat.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def services_up?
 
   def memcached_up?
     host = ENV["MEMCACHE_SERVERS"]
-    memcached_client = Dalli::Client.new("#{host}:11211")
+    memcached_client = Dalli::Client.new(ENV["MEMCACHE_SERVERS"])
```
Comment (jrhoads) [inline app/models/heartbeat.rb:22]: This is the same except that it no longer specifies the port?

File: bin/setup [hunk 2]
```diff
@@ -15,7 +15,7 @@ chdir APP_ROOT do
   # Add necessary setup steps to this file.
 
   puts '== Installing dependencies =='
-  system! 'gem install bundler --conservative'
+  system! 'gem install bundle -v 2.5.6 r --conservative'
```
Comment (jrhoads) [inline bin/setup:18]: Is the `r` separated from the `bundle`?

File: Gemfile.lock [hunk 3]
```diff
@@ -182,10 +183,9 @@ GEM
       mime-types
       mimemagic (~> 0.3.0)
       terrapin (~> 0.6.0)
-    listen (3.1.5)
+    listen (3.0.8)
```
Comment (jrhoads) [inline Gemfile.lock:186]: Seems strange that this would downgrade the version

File: app/models/heartbeat.rb (hunk 1 above)
Comment (wendelfabianchinsamy) [reply app/models/heartbeat.rb:22]: the result is exactly the same. i'm following the convention used in lupo which uses the environment variable. the environment variable does have the port.

File: bin/setup (hunk 2 above)
Comment (wendelfabianchinsamy) [reply bin/setup:18]: thank you!

File: app/models/heartbeat.rb (hunk 1 above)
Comment (jrhoads) [reply app/models/heartbeat.rb:22]: Could get rid of line 21

File: Gemfile.lock (hunk 3 above)
Comment (wendelfabianchinsamy) [reply Gemfile.lock:186]: yes it is however listen is a development dependency that plays no role in the production environment.

**Review by jrhoads — APPROVED** [review]

**Pull request merged**